test_conformance/api/test_queries.cpp

```diff
@@ -796,6 +796,72 @@ REGISTER_TEST(get_device_info)
     return 0;
 }
 
+REGISTER_TEST(get_device_info_comparability)
+{
+    int error = CL_SUCCESS;
+
+    // comparability test for CL_DEVICE_PLATFORM
+    // 1. find platform related to device without using query
+    cl_uint num_platforms = 0;
+    error = clGetPlatformIDs(16, nullptr, &num_platforms);
+    test_error(error, "clGetPlatformIDs failed");
+
+    std::vector<cl_platform_id> platforms(num_platforms);
+
+    error = clGetPlatformIDs(num_platforms, platforms.data(), &num_platforms);
+    test_error(error, "clGetPlatformIDs failed");
+
+    cl_uint num_devices = 0;
+    cl_platform_id comp_platform = nullptr;
+    for (int p = 0; p < (int)num_platforms && comp_platform == nullptr; p++)
+    {
+        error = clGetDeviceIDs(platforms[p], CL_DEVICE_TYPE_ALL, 0, nullptr,
+                               &num_devices);
+        if (error != CL_SUCCESS || num_devices == 0) continue;
+
+        std::vector<cl_device_id> devices(num_devices);
+        error = clGetDeviceIDs(platforms[p], CL_DEVICE_TYPE_ALL, num_devices,
+                               devices.data(), nullptr);
+        if (error != CL_SUCCESS) continue;
+
+        // find correct device
+        for (auto did : devices)
+        {
+            if (did == device)
+            {
+                comp_platform = platforms[p];
+                break;
+            }
+        }
+    }
+
+    // fallback path
+    if (comp_platform == nullptr)
+    {
+        char *env_mode = getenv("CL_PLATFORM_INDEX");
+        if (env_mode != nullptr)
+        {
+            int choosen_platform_index = atoi(env_mode);
+            if (choosen_platform_index < static_cast<int>(platforms.size()))
+                comp_platform = platforms[choosen_platform_index];
+        }
+    }
```

Comment:
I don't think we need this "fallback path".  We've already searched through all of the platforms on the system, so we shouldn't find it if we look again at the platform described by `CL_PLATFORM_INDEX`.  We can just remove this entire code block.

Reply:
Done.